src/main/java/de/chojo/repbot/dao/access/guild/settings/sub/Thanking.java

```diff
@@ -0,0 +1,140 @@
+package de.chojo.repbot.dao.access.guild.settings.sub;
+
+import de.chojo.jdautil.consumer.ThrowingConsumer;
+import de.chojo.repbot.dao.access.guild.settings.Settings;
+import de.chojo.repbot.dao.access.guild.settings.sub.thanking.Channels;
+import de.chojo.repbot.dao.access.guild.settings.sub.thanking.DonorRoles;
+import de.chojo.repbot.dao.access.guild.settings.sub.thanking.Reactions;
+import de.chojo.repbot.dao.access.guild.settings.sub.thanking.ReceiverRoles;
+import de.chojo.repbot.dao.access.guild.settings.sub.thanking.Thankwords;
+import de.chojo.repbot.dao.components.GuildHolder;
+import de.chojo.sqlutil.base.QueryFactoryHolder;
+import de.chojo.sqlutil.wrapper.ParamBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+
+public class Thanking extends QueryFactoryHolder implements GuildHolder {
+    private final String mainReaction;
+    private final Settings settings;
+    private final boolean channelWhitelist;
+
+    private Channels channels;
+    private DonorRoles donorRoles;
+    private ReceiverRoles receiverRoles;
+    private Reactions reactions;
+    private Thankwords thankwords;
+
+    public Thanking(Settings settings) {
+        this(settings, null, true);
+    }
+
+    public Thanking(Settings settings, String mainReaction, boolean channelWhitelist) {
+        super(settings);
+        this.settings = settings;
+        this.mainReaction = mainReaction == null ? "🏅" : mainReaction;
```

Comment:
I think it is better to work without null here. Or is there any reason to pass null here which needs to get handled?
```suggestion
        this(settings, "🏅", true);
    }

    public Thanking(Settings settings, String mainReaction, boolean channelWhitelist) {
        super(settings);
        this.settings = settings;
        this.mainReaction = mainReaction;
```

Reply:
The constructor is used by the build function where the reaction actually can be null
Actually no. Probably just an artifact. Gonna pass the default value in there.
I also moved the default value to a constant.

Additionally I added the default value to the column to ensure it is always set.